wordpress-mobile/WordPress-Android
# Issue/base appbar styling

This PR sets the base for future AppBar redesign work. It's focused on modifying global theme values and making sure the My Site, Reader, and Notification screens look and behave as expected (excluding nested screens for now).

The big item in this PR is logic related to making AppBars "Lif on Scroll" working with Fragments located in ViewPager. To correctly change the AppBar elevation it needs to know which view is being scrolled. By default, AppBar looks for the first scrollable view inside parent CoordinatorLayout, alternatively, we can explicitly tell AppBar ID of the scrollable view it should observe. Neither options work too well with multiple nested fragments inside ViewPager that all use the same layout with the same ID for scrollable view. For this reason, we need to tell the app

My approach is to set a unique ID to a scrollable view and send it to the parent fragment or activity after the fragment in ViewPager becomes active (when it's selected and you can interact with it). Changing ID of the view after we get a reference to it in Fragment should not be an issue, according to my research and testing. This also works with View Binding of Kotling Extensions - on the first time you get a view by ID it's stored in cache map with the original ID as a key, and on subsequent requests, the cached reference is used - you can confirm this by decompiling the kotlin bytecode.
It tried to extract the code and make it as easy as possible to implement, It works, but I'm open to any suggestions 🙏 

Note about Notification List - because of the `CollapsingToolbarLayout` with specific scroll flags we can set the `liftOnScroll` to `false` and `AppBar` will figure out the elevation based on `CollapsingToolbarLayout` behavior.

Another big change is in the behavior of reader AppBar - now it scrolls away to give you more space for content:

On the visual side, I had no redlines, so I eyeballed the look. @osullivanchris please let me know if anything looks out of place :)

![appbar](https://user-images.githubusercontent.com/728822/88758180-aeca3b80-d11c-11ea-8429-25c59ed1dc88.png)

To test:
- Check My Site, Reader and Notifications screens and make sure they work as expected.
- Make sure "lift on scroll" works when you scroll the content. In Light mode you will see shadow below the AppBar, find Dark Mode AppBars elevation will change it will become lighter/flush on scroll.
- Make sure gravater at My Site screen changes size with scroll.
- Make sure divider between tabs and content disappears on scroll.

PR submission checklist:
